### PR TITLE
BTRX - 818 - Session handling

### PR DESCRIPTION
**Pull request by LeoForconesi**
## Addresses
[BTRX-818: ](https://broadinstitute.atlassian.net/browse/BTRX-818)

---

- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

**Review by tadeork — APPROVED** [review]
:+1:

File: grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy [hunk 1]
```diff
@@ -147,4 +154,22 @@ class DataUseLetterController implements ExceptionHandler, UserInfo {
         render(text: matches as JSON, contentType: "application/json")
     }
 
+
+    def findByUUID() {
+        String uid = params.uuid
```
Comment (dags) [inline grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy]: uid used once. Use params.uuid instead.

File: src/main/webapp/projectContainer/SubmissionForm.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ const SubmissionForm = hh(class SubmissionForm extends Component {
       };
 
       ProjectMigration.saveSubmission(submissionData, this.state.documents, this.state.params.submissionId).then(resp => {
-        this.backToProject();
+        this.backToProject();
```
Comment (dags) [inline src/main/webapp/projectContainer/SubmissionForm.js]: remove extra spaces

Comment (LeoForconesi) [reply src/main/webapp/projectContainer/SubmissionForm.js]: done!

**Review by dags — APPROVED** [review]

File: grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy [hunk 3]
```diff
@@ -147,4 +154,21 @@ class DataUseLetterController implements ExceptionHandler, UserInfo {
...
+            User user = userService.findUser(dul.getCreator())
+            Issue issue = queryService.findByKey(dul.getConsentGroupKey())
+            consent.put("dataManagerName", user?.getDisplayName())
+            consent.put("dataManagerEmail", user?.getEmailAddress())
+            consent.put("consentGroupKey", issue?.getProjectKey())
+            consent.put(IssueExtraProperty.SUMMARY, issue?.getSummary())
+            consent.put(IssueExtraProperty.PROTOCOL, IssueExtraProperty.findByProjectKeyAndName(dul.getConsentGroupKey(), IssueExtraProperty.PROTOCOL).getValue())
+        }
+        response.status = 200
+        render([consent: consent] as JSON)
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/api/DataUseLetterController.groovy:171]: Few details to point out in this method.
1. We shouldn't be using GORM's `IssueExtraProperty.findByProjectKeyAndName` here, better to use raw sql.
2. All of this data can be easily queried via raw sql.
3. `consent` doesn't seem like the right name here. We're returning an ad-hoc collection of user, project, and consent information. Given that this looks like a useful construct, it should be a full model object that can be returned by `QueryService`
4. Nitpicking: spacing around the `if` condition.

File: src/main/webapp/components/TopNavigationMenu.js [hunk 4]
```diff
@@ -114,31 +113,27 @@ const TopNavigationMenu = hh(class TopNavigationMenu extends Component {
...
     Storage.clearStorage();
-    if (window.gapi.auth2 != undefined) {
-      let auth2 = window.gapi.auth2.getAuthInstance();
-      await auth2.signOut();
-      await User.signOut();
-    }
-    this.setState({
+    component.isBroad = null;
+    component.isAdmin = null;
+    component.isViewer = null;
```
Comment (rushtong) [inline src/main/webapp/components/TopNavigationMenu.js:131]: I'm seeing this in a lot of components. We should look into refactoring these into a user check or into a common HOC.

**Pull request merged**